src/tcpconnect/main.rs

```diff
@@ -0,0 +1,270 @@
+            .number_of_values(1)
+            .takes_value(true)
+        )
+        .arg(
+            Arg::with_name("p")
+            .long("pid")
+            .help("trace this PID only")
+            .value_name("PID")
+            .number_of_values(1)
+            .takes_value(true)
+        )
+        .arg(
+            Arg::with_name("ebpf")
+            .takes_value(false)
+        )
+        .arg(
+            Arg::with_name("P")
+            .long("port")
+            .help("comma-separated list of destination ports to trace.")
+            .value_name("PORT")
+            .min_values(1)
+            .required(false)
+            .takes_value(true)
+        )
+        .get_matches();
+        // first thing is to swap code
+        bpf_text = bpf_text.replace("IPV4_CODE", ipv4_trace_code);
+        bpf_text = bpf_text.replace("IPV6_CODE", ipv6_trace_code);
+        // handle pids
+        if let Some(p) = matches.value_of("p") {
+            // trace pid
+            let pid = p.parse::<u32>().expect("pid should be a non-negative integer");
+            bpf_text = {
+                let uuids = format!("if (pid != {}) {{ return 0; }}", pid);
+                bpf_text.replace("FILTER_PID", uuids.as_str())
+            };
+        }
+        // handle ports 
+        let port_listings = matches.values_of("P");
+        if port_listings.is_some() {
+            let mut port_listings_raw = port_listings.expect("comma separated listing of ports, e.g. 1,2,3,80");
+            let mut ports_conditions: Vec<String> = Vec::new();
+            loop{
+                match port_listings_raw.next() {
+                    Some(port) => {
+                        let port_u16 = port.parse::<u16>().expect("port number between 1-65535");
+                        let port_ntohs = ntohs(port_u16);
+                        let port_condition = format!("dport != {}", port_ntohs);
+                        ports_conditions.push(port_condition);
+                    },  
+                    None => {break;}
+                }
+            }
+            let port_contition_statement = ports_conditions.join(" && ");
+            bpf_text = bpf_text.replace("FILTER_PORT", format!("if ({}) {{ currsock.delete(&pid); return 0; }}", port_contition_statement).as_str());
+        }
+        // handle uid
+        if let Some(u) = matches.value_of("u") {
+            // trace uuid
+            let uuid = u.parse::<u32>().expect("uuid value should be an positive integer");
+            bpf_text = {
+                let uuid_set = format!("if (uid != {}) {{ return 0; }}",uuid);
+                bpf_text.replace("FILTER_UID", uuid_set.as_str())
+            };
+        }
+        // replace remaining unset template flags, if any
+        bpf_text = bpf_text.replace("FILTER_PID", "");
+        bpf_text = bpf_text.replace("FILTER_PORT", "");
+        bpf_text = bpf_text.replace("FILTER_UID", "");
+
+        let ebpf = matches.is_present("ebpf");
```

Comment:
I think we can probably get rid of this

Reply:
I got rid of the epbf flag since we can remove debug functionality.